apps/ecommerce/providers/shopify/src/controller/ResourceTypeController.ts

```diff
@@ -1,53 +1,72 @@
-import { resourceTypeSchema } from '../services/resourceType/ResourceTypeService';
+import {
+  appInstallConfig,
+  resourceTypeConfig,
+  shopifySchema,
+} from '../services/resourceType/ResourceTypeService';
 import { extractShopifyClientConfigParams } from '../helpers/ClientHelper';
 import { Router } from 'express';
 import { Request, Response } from 'express';
 import { fetchProduct, fetchProducts } from '../handlers/ShopifyHandler';
 
 const ResourceTypeController = Router();
 
-const ping = async (req: Request, res: Response) => {
-  return res.status(200).send('healthy!');
+const getAppInstallConfig = async (req: Request, res: Response) => {
+  return res.status(200).send(appInstallConfig);
 };
 
-const getResourceType = async (req: Request, res: Response) => {
-  return res.status(200).send(resourceTypeSchema);
+const getResourceTypeConfig = async (req: Request, res: Response) => {
+  return res.status(200).send(resourceTypeConfig);
 };
 
 const getResourceTypeSchema = async (req: Request, res: Response) => {
-  return res.status(200).send('getResourceTypeSchema call successful');
+  return res.status(200).send(shopifySchema);
 };
 
 const getResources = async (req: Request, res: Response) => {
   // TODO: HANDLE PAGINATION, QUERYING, NOR PAGINATION
-  const storefrontAccessToken = JSON.parse(JSON.stringify(req.header('x-storefront-access-token')));
-  const shopName = JSON.parse(JSON.stringify(req.header('x-shop-name')));
+
+  // Have parse twice, one because we stringify the header here, second because the header in the request is also stringified,
+  // this is because the x-data-provider-parameters header represents an object which needs to be stringified before being sent as a request
+  const providerHeaders = JSON.parse(
+    JSON.parse(JSON.stringify(req.header('x-data-provider-parameters')))
```

Comment:
They need to be bundled as one because the proxy can't make any assumptions about who the headers are for. For example what if Magento calls their shopName -> storeName instead? We need to bundle these installation parameters under a generic header and pass that along to the Provider who will know how to destructure the header object.